cranelift/filetests/filetests/isa/riscv64/constants.clif

```diff
@@ -96,17 +96,17 @@ block0:
 
 ; VCode:
 ; block0:
-;   ld a0,[const(0)]
+;   lui a0,16
+;   addi a2,a0,-1
+;   slli a0,a2,32
 ;   ret
 ;
 ; Disassembled:
 ; block0: ; offset 0x0
-;   auipc a0, 0
-;   ld a0, 0x10(a0)
+;   lui a0, 0x10
+;   addi a2, a0, -1
+;   slli a0, a2, 0x20
```

Comment:
`lui` accepts 20 bits immediates so this could probably be optimized further:

```asm
   lui a0, 0xffff
   slli a0, a0, 0x14
```

Reply:
Oh right, that's a good catch! Since we always test materializing the immediate at position 0, there could be another shift amount that would produce a better immediate generation sequence.

I wonder how we could better match these other cases.

Reply:
Opened #7139